src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3241,18 +3242,37 @@ static size_t large_page_init_decide_size() {
     return 0;
   }
 
-#if defined(IA32) || defined(AMD64)
-  if (size > 4*M || LargePageSizeInBytes > 4*M) {
-    WARN("JVM cannot use large pages bigger than 4mb.");
-    return 0;
-  }
+  // Check if OS version is 11 or greater
```

Comment:
This comment is not accurate (you're checking for Window Server 2022 or greater as well). I'm also not sure it's needed, as the function calls are pretty self-explanatory.